src/manifest.rs

```diff
@@ -99,61 +99,102 @@ pub(crate) struct LintTable {
+                    0,
+                    QueryOverride {
+                        lint_level: Some(lint_level),
+                        required_update: None,
+                    },
+                ),
+                OverrideConfig::Both {
+                    level,
+                    required_update,
+                    priority,
+                } => (
+                    priority,
+                    QueryOverride {
+                        lint_level: Some(level),
+                        required_update: Some(required_update),
+                    },
+                ),
+                OverrideConfig::LintLevel { level, priority } => (
+                    priority,
+                    QueryOverride {
+                        lint_level: Some(level),
+                        required_update: None,
+                    },
+                ),
+                OverrideConfig::RequiredUpdate {
+                    required_update,
+                    priority,
+                } => (
+                    priority,
+                    QueryOverride {
+                        lint_level: None,
+                        required_update: Some(required_update),
+                    },
+                ),
+            };
+
+            map.entry(priority).or_default().insert(id, overrides);
+        }
+
+        // this will be sorted by key `priority` in ascending order.
+        // to get the stack semantics of later elements take priority,
+        // we need to reverse this.
+        map.into_values().rev().collect()
     }
 }
 
 /// Different valid representations of a [`QueryOverride`] in the Cargo.toml configuration table
 #[derive(Debug, Clone, Deserialize)]
 #[serde(untagged)]
 pub(crate) enum OverrideConfig {
-    /// Specify members by name, e.g.
+    /// Specify both lint level and required update by name, e.g.
     /// `lint_name = { level = "deny", required-update = "major" }
-    /// Any omitted members will default to `None`
     #[serde(rename_all = "kebab-case")]
-    Structure {
+    Both {
+        level: LintLevel,
+        required_update: RequiredSemverUpdate,
+        /// The priority for this configuration.  If there are multiple entries that
+        /// configure a lint (e.g., a lint group containing a lint and the lint itself),
+        /// the configuration entry with the **lowest** priority takes precedence.
+        /// The default value, if omitted, is 0.
         #[serde(default)]
-        level: Option<LintLevel>,
+        priority: i64,
+    },
+    /// Specify just lint level by name, with optional priority.
+    /// `lint_name = { level = "deny" }
+    #[serde(rename_all = "kebab-case")]
+    LintLevel {
+        level: LintLevel,
         #[serde(default)]
-        required_update: Option<RequiredSemverUpdate>,
+        priority: i64,
+    },
+    /// Specify just required update by name, with optional priority.
+    /// `lint_name = { level = "deny" }
```

Comment:
This appears to be accidental copy-pasta since it references level instead of required update.